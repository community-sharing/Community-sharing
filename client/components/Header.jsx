import React from 'react'
import { connect } from 'react-redux'
import { menuNavigation } from '../actions'


import NavigationMenu from './NavigationMenu.jsx'

let Header = (props) => (

  <div className='header-menu-container'>

    <div className='header-container'>
      <div className='hambuger-container' onClick={ () => openMenu(props) }>
        <i className="fa fa-bars" aria-hidden="true"></i>
      </div>

      <div className='title-container'>
        <p className='site-title'>Community Sharing</p>
      </div>

      <div className='login-button-container'>
        <i className="fa fa-user" aria-hidden="true"></i>
      </div>
    </div>

     { props.menuState ? <NavigationMenu /> : "" }

  </div>
)

function mapStateToProps(state){
  return {
    dispatch: state.dispatch,
    menuState: state.menuState
  }
}

export default connect(mapStateToProps)(Header)

function openMenu(props) {
  props.dispatch(menuNavigation())
}
