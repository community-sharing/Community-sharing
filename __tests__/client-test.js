import React from 'react';
import renderer from 'react-test-renderer';
import Order from '../client/components/Order.jsx';
import {Header} from '../client/components/Header.jsx';

describe('Header (Snapshot)', () => {
  it('The Header renders', () => {
    const component = renderer.create(<Header />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Order (Snapshot)', () => {
  it('The Order renders', () => {
    const component = renderer.create(<Order />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
