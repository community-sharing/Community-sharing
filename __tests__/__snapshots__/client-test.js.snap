// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header (Snapshot) The Header renders 1`] = `
<div
  className="header-menu-container"
>
  <div
    className="header-container"
  >
    <div
      className="hambuger-container"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-bars"
      />
    </div>
    <div
      className="title-container"
    >
      <p
        className="site-title"
      >
        Community Sharing
      </p>
    </div>
    <div
      className="login-button-container"
    >
      <i
        aria-hidden="true"
        className="fa fa-user"
      />
    </div>
  </div>
  
</div>
`;

exports[`Order (Snapshot) The Order renders 1`] = `
<div
  class="Order"
>
  <h1>
    Request Approved - Transaction Receipt
  </h1>
  <h3>
    Item Details
  </h3>
  <ul>
    <li>
      Item: Lawn Mower
    </li>
    <li>
      Pick Up Date: 01-01-2015
    </li>
    <li>
      Drop Off Date: 03-01-2015
    </li>
  </ul>
  <h3>
    Lender Details
  </h3>
  <ul>
    <li>
      Name: Dan Ba-la-kay
    </li>
    <li>
      Phone: 021120021
    </li>
    <li>
      email: dblake@gunit.com
    </li>
  </ul>
  <h3>
    Borrower Details
  </h3>
  <ul>
    <li>
      Name: Pou Pou
    </li>
    <li>
      Phone: 021120021
    </li>
    <li>
      email: pou@yeahboiii.com
    </li>
  </ul>
</div>
`;
